bundler/lib/bundler.rb

```diff
@@ -503,12 +503,19 @@ def mkdir_p(path)
     def which(executable)
       if File.file?(executable) && File.executable?(executable)
         executable
-      elsif paths = ENV["PATH"]
+      elsif (paths = ENV["PATH"])
         quote = '"'
         paths.split(File::PATH_SEPARATOR).find do |path|
           path = path[1..-2] if path.start_with?(quote) && path.end_with?(quote)
-          executable_path = File.expand_path(executable, path)
-          return executable_path if File.file?(executable_path) && File.executable?(executable_path)
+          if Gem.win_platform? then
+            executable_path = File.expand_path(executable+".bat", path)
```

Comment:
Wonder if we need to check other exts

Reply:
I think for this case using `which` is not needed.

We can just check ruby installation path ( `ruby -rrbconfig -e 'puts RbConfig::CONFIG["prefix"]'` ) + `bin` and current folder + `bin`

Reply:
`bundle exec` should be able to run any arbitrary executable, not just executables in the same directory where ruby is installed.